### PR TITLE
Fix duplicate user name bug

### PR DESCRIPTION
**Pull request by bruh-9000**
https://github.com/moddio/moddio2/pull/675

### CODE REVIEW

**Comment by m0dE** [comment]
please provide rationale behind this PR. I can see some reasons why, but I'd like to hear from you. also plz do this for all future PRs. @bruh-9000

**Comment by m0dE** [comment]
@sloont please see if you can implement this in clusterServer logic (or should we?)

**Comment by bruh-9000** [comment]
@m0dE rationale is that if there are multiple users with the same name, scripts that utilize player names may confuse them to be the same player and have issues. Rare, but possible, I've seen multiple users with the same name before.

File: server/ServerNetworkEvents.js [hunk 1]
```diff
@@ -61,6 +61,14 @@ var ServerNetworkEvents = {
 	},
 
 	_onJoinGame: function (data, clientId) {
+		var existing_names = taro.$$('player').filter(player => player._stats.controlledBy === 'human').map(player => player._stats.username.substring('user'.length))
```
Comment (sloont) [inline server/ServerNetworkEvents.js:64]: this needs to be broken down into more readable code. it's too smart for its own good. on that note, this really needs comments explaining what is happening. chaining two array methods and internally using a string method unnecessarily is hard to follow.

also describe what this array is meant to look like after all of this processing--im guessing its an array of multiple `undefined` and guest user numbers

File: server/ServerNetworkEvents.js [hunk 2]
```diff
@@ -61,6 +61,14 @@ var ServerNetworkEvents = {
...
+
+		if (existing_names.includes(data.number)) {
+			do {
```
Comment (sloont) [inline server/ServerNetworkEvents.js:67]: this is good though. comment though pls

**Review by sloont — CHANGES_REQUESTED** [review]
